src/unicorn/builder.rs

```diff
@@ -531,6 +531,23 @@ impl ModelBuilder {
         self.reg_flow_ite(itype.rd(), sra_node);
     }
 
+    fn model_sllw(&mut self, rtype: RType) {
+        let thirtytwo = self.new_const(32);
+
+        // Only the low 4 bits of rs2 are considered for the shift amount.
+        let mask_node = self.new_const(0xf); // TODO: Make this a global constant.
```

Comment:
Why only 4 bits? I think one has to use 5 bits to express the full range of valid shift amounts between 0 and 31. The syntax from the spec saying n[0:4] is inclusive and means bits [0,1,2,3,4] as far as I know. The mask should be `0x1f` instead I think.